R/render.R

```diff
@@ -22,14 +22,24 @@ create_prevelance_renderer <- function(
   renderer
   ) {
   function(timestep) {
+    
     asymptomatic <- state$get_index_of('A')
-    prob <- probability_of_detection(
-      get_age(birth$get_values(asymptomatic), timestep),
-      immunity$get_values(asymptomatic),
-      parameters
-    )
-    asymptomatic_detected <- bitset_at(asymptomatic, bernoulli_multi_p(prob))
-
+    
+    if(parameters$parasite == "falciparum"){
+      prob <- probability_of_detection(
+        get_age(birth$get_values(asymptomatic), timestep),
+        immunity$get_values(asymptomatic),
+        parameters
+      )
+      asymptomatic_detected <- bitset_at(asymptomatic, bernoulli_multi_p(prob))
+      
+    } else if (parameters$parasite == "vivax"){
+      # The vivax model defines asymptomatic infections as being detectable by
+      # light microscopy
+      prob <- rep(1, asymptomatic$size())
```

Comment:
I kinda love this...

But we should really avoid this prob stuff entirely when doing the vivax model (i.e. p_detect = n_detect)

Reply:
For sure - I left this in as something of a cry for help.

I have a couple of wider questions about what's getting rendered and when. I think there are two places where age rendering can currently occur? One in create_age_group_renderer as expected, but then one in create_prevelance_renderer consistent with the prevalence age group. The documentation for create_prevelance_renderer also says it's rendering prevalence of severe malaria, but doesn't look like that's happening in this part of the code...? (The run_simulation documentation also says that n_severe is an output, but I'm unsure of where that happens). I wonder if this function was intended to show severe prevalence, rather than within an age group. I do think that age rendering should probably only happen once. Then for P. vivax, I can report the number in A, D or Tr compartments for a comparable output...

Any thoughts on that would be appreciated!

Reply:
Ah yes! We used to render severe prevalence but that was a mistake, I must've removed the code but not the documentation. Made an issue #262 

> One in create_age_group_renderer

I actually didn't know this was a thing 😆 

But it looks like they are separate features. One is for prevalence statistics (including associated age counts) and the other is for if you want age counts. I'm assuming the latter was introduced so that people could validate their demography configuration or calculate age specific intervention coverage statistics if they wanted to.

> I do think that age rendering should probably only happen once.

I actually disagree. I think that if the user parameterises age counts twice (once with prevalence_rendering and again with age_group_rendering) we should do it twice.

 > Then for P. vivax, I can report the number in A, D or Tr compartments for a comparable output...

I don't follow here sorry...

Reply:
I realised yesterday that there are actually three places where age rendering takes place: the prevalence function, the incidence function and the age-rendering function (which means that it's consistent to have age rendering within incidence/prevalence rendering and I now agree with you). But I did also notice that in prevalence and incidence the age outputs are both n_lower_upper, while in the age-rendering the output is n_age_lower_upper. I wonder if the outputs should labelled be the same?

> I don't follow here sorry...

The falciparum model reports n_detect and p_detect in prevalence (LM detectable prevalence, which are clinically detected (Tr or D) or asymptomatic (A)). In vivax the asymptomatic compartment is (by definition) LM detectable, so there is no need to calculate p_detect for vivax. It would still be good to output a comparable metric for vivax, so maybe I'll just output n_detect for both parasites and only p_detect for falciparum. Does that make sense?

Reply:
@giovannic @pwinskill 
I've just been checking through some of the previous suggestions and this is what I propose in terms of the n_detect and p_detect e.g.,
> Perhaps this is an opportunity to make the columns more specific, e.g. with n_detect_microscopy_ for falciparum and n_detect_LM_ and n_detect_PCR_ for vivax runs?

Current situation:
- n_detect_ (D + Tr + A)
- p_detect_ (D + Tr + prob(A), where prob due to detectability immunity)

Previously suggested:
- n_detect_LM_ (for Pv, D + Tr + A)
- n_detect_PCR_ (for Pv, D + Tr + A + U)
- n_detect_microscopy (although this seems equivalent to n_detect_LM_, so could condense into one?)

So I suggest:
n_detect_ → n_detect_lm_ (pf and pv: D + Tr + A)
p_detect_ → p_detect_lm_ (pf only: D + Tr + prob(A), where prob due to detectability immunity)
n_detect_pcr_ (new: pf and pv: D + Tr + A + U)

How does that sound? Any thoughts?

Reply:
Pete just gave a good explanation for why it's good to have a separate age renderer which is named differently, so please ignore me on that point!